src/Microsoft.AspNet.Diagnostics/Views/CompilationErrorPage.cs

```diff
@@ -95,7 +95,7 @@ public CompilationErrorPage()
 #line hidden
             WriteLiteral(": ");
 #line 30 "CompilationErrorPage.cshtml"
-                                                                          Output.Write(WebUtility.HtmlEncode(errorDetail.Error.Message).Replace("\r\n", "<br/>").Replace("\r", "<br/>").Replace("\n", "<br/>")); 
+                                                                          Output.Write(HtmlEncoder.HtmlEncode(errorDetail.Error.Message).Replace("\r\n", "<br/>").Replace("\r", "<br/>").Replace("\n", "<br/>")); 
```

Comment:
Line breaks must be stripped before passing the string through the encoder, as the encoder could replace the line breaks with whatever representation it wishes. See https://github.com/aspnet/Helios/blob/dev/src/Microsoft.AspNet.Loader.IIS/Util/ErrorHandlingExtensions.cs#L350 for an example.